indigo_app/forms.py

```diff
@@ -952,10 +988,54 @@ def is_repealed_work_saved(self):
     def save(self):
         work = self.cleaned_data['repealed_work']
         if self.cleaned_data.get('DELETE'):
-            work.repealed_by = None
-            work.repealed_date = None
-            work.save()
+            if work.repealed_by == self.work:
+                work.repealed_by = None
+                work.repealed_date = None
+                work.save_with_revision(user=self.cleaned_data['updated_by_user'])
         else:
             work.repealed_by = self.work
             work.repealed_date = self.cleaned_data['repealed_date']
-            work.save()
+            work.save_with_revision(user=self.cleaned_data['updated_by_user'])
+
+
+RepealMadeBaseFormSet = formset_factory(
+    RepealMadeForm,
+    extra=0,
+    can_delete=True,
+)
+
+
+class AmendmentForm(forms.Form):
+    amending_work = forms.ModelChoiceField(queryset=Work.objects)
+    date = forms.DateField(widget=forms.DateInput(attrs={'type': 'date'}), required=False)
+    updated_by_user = forms.ModelChoiceField(queryset=User.objects, required=False)
+    created_by_user = forms.ModelChoiceField(queryset=User.objects, required=False)
```

Comment:
I think updated_by_user and created_by_user can come out

Reply:
Thought they are null=True, they are used elsewhere in the code, I think to create the actions stream

Reply:
That's 100% right yes — I was still hoping you'd just be able to pass the user straight, but they're definitely needed if you can't